app/assets/javascripts/thyme.js

```diff
@@ -0,0 +1,1092 @@
+
+  idleHideControlBar();
+
+  // if mouse is moved or screen is toiched, show control bar
+  video.addEventListener('mouseover', showControlBar, false);
+  video.addEventListener('mousemove', showControlBar, false);
+  video.addEventListener('touchstart', showControlBar, false);
+
+  // Event listener for the play/pause button
+  playButton.addEventListener('click', function() {
+    if (video.paused == true) {
+      video.play();
+    } else {
+      video.pause();
+    }
+  });
+
+  video.onplay = function() {
+    playButton.innerHTML = 'pause';
+  };
+
+  video.onpause = function() {
+    playButton.innerHTML = 'play_arrow';
+  };
+
+  // Event listener for the mute button
+  muteButton.addEventListener('click', function() {
+    if (video.muted == false) {
+      video.muted = true;
+      muteButton.innerHTML = 'volume_off';
+    } else {
+      video.muted = false;
+      muteButton.innerHTML = 'volume_up';
+    }
+  });
+
+  // Event handler for the plusTen button
+  plusTenButton.addEventListener('click', function() {
+    video.currentTime = Math.min(video.currentTime + 10, video.duration);
+  });
+
+  // Event handler for the minusTen button
+  minusTenButton.addEventListener('click', function() {
+    video.currentTime = Math.max(video.currentTime - 10, 0);
+  });
+
+  // Event handler for the nextChapter button
+  nextChapterButton.addEventListener('click', function() {
+    var next = nextChapterStart(video.currentTime);
+    if (next != null) {
+      video.currentTime = nextChapterStart(video.currentTime);
+    }
+  });
+
+  // Event handler for the previousChapter button
+  previousChapterButton.addEventListener('click', function() {
+    var previous = previousChapterStart(video.currentTime);
+    if (previous != null) {
+      video.currentTime = previousChapterStart(video.currentTime);
+    }
+  });
+
+  // Event handler for the emergency button
+  emergencyButton.addEventListener('click', function() {
+    video.pause();
+    $.ajax(Routes.new_annotation_path(), {
+      type: 'GET',
+      dataType: 'script',
+      data: {
+        total_seconds: video.currentTime,
+        mediumId: thyme.dataset.medium
+      }
+    });
+    // When the modal opens, all key listeners must be
+    // deactivated until the modal gets closed again
+    lockKeyListeners = true;
+    $('#annotation-modal').on('hidden.bs.modal', function() {
+      lockKeyListeners = false;
+    });
+  });
+
+  if (annotationsToggle !== null) {
+    annotationsToggle.addEventListener('click', function() {
+      updateMarkers();
+    });
+  }
+
+  // Update annotations after submitting the annotations form
+  $(document).on('click', '#submit-button', function() {
+    /* NOTE:
+       Updating might take some time on the backend,
+       so I added a slight delay.
+       I couldn't think of an easy way to let the script
+       wait for the update to complete (as with the delete button),
+       but it might be possible! */
```

Comment:
You can send a JS file to the frontend when the controller is finished. This file could contain `updateMarkers()`. See how I realized something similar for the feedback button:
- I have a `create` method [here](https://github.com/MaMpf-HD/mampf/pull/527/files#diff-2b0bba658ec28b34a9c6bea0cacecc10022b929dd8aed5df15d92e1d93856e86R4-R7).
- Then, correspondingly to the name of the controller method, a `create.js.erb` file [here](https://github.com/MaMpf-HD/mampf/pull/527/files#diff-cee5fcef04cf78d4b0883370c679f768eb2d1ae4b8a4561674ecefd9dfb414a3) that is sent to the frontend.